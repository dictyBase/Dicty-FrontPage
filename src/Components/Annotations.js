// @flow
import React from "react"
import { Box} from "grid-styled"
import FontAwsome from "react-fontawesome"
import styled from "styled-components"

import {
  AnnotationTitle,
  AnnotationSubTitle,
  AnnotationHeader,
  ListBox,
  AnnotationListItems,
  AnnotationContainer,
  MoreLink
} from "./styles"


const Container = styled.div`
display: flex;
flex-direction: row;
row-wrap: wrap;
width: 90%;
margin: auto;
`


const Annotations = (props) => {
  const genelist = props.annotations.genes.map((gene, index) =>
    <AnnotationListItems key={index}>
        {gene}
    </AnnotationListItems>,
  )

  const paperlist = props.annotations.papers.map((paper, index) =>
    <AnnotationListItems key={index}>
        {paper}
    </AnnotationListItems>,
  )

  return (
      <AnnotationContainer>
            <AnnotationHeader>
                <FontAwsome name="pencil fa-md" />
                <AnnotationTitle>
                  RECENT ANNOTATIONS
                </AnnotationTitle>
            </AnnotationHeader>
            <Container>
                <Box px={2} py={1} mt={-2} width={1 / 2}>
                    <AnnotationSubTitle>
                      Genes
                    </AnnotationSubTitle>
                    <ListBox margintop="0px" padbottom="0px">
                      {genelist}
                    </ListBox>  
                    <MoreLink padbottom="0px">
                      <FontAwsome name="plus fa-xs" />
                    </MoreLink>         
                </Box> 
                <Box px={2} py={1} mt={-2} width={1 / 2}>
                    <AnnotationSubTitle>
                      Papers
                    </AnnotationSubTitle>                
                    <ListBox margintop="0px" padbottom="0px">
                      {paperlist}
                    </ListBox> 
                    <MoreLink padbottom="0px">
                      <FontAwsome name="plus fa-xs" />
                    </MoreLink>                     
                </Box> 
            </Container>
      </AnnotationContainer>
  )
}

export default Annotations
