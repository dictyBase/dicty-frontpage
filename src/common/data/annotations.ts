export default {
  genes: ["erkB", "kif8", "cln3", "tpp1"],
  papers: ["30612939", "31067156", "30771446", "31100984"],
}
