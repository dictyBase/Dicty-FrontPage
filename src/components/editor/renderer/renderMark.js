// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
