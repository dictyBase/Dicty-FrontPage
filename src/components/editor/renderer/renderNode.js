// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
