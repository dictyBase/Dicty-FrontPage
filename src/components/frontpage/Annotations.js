// @flow
import React from "react"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import FontAwesome from "react-fontawesome"

import {
  AnnotationTitle,
  AnnotationSubTitle,
  AnnotationHeader,
  ListBox,
  AnnotationListItems,
  AnnotationContainer,
  MoreLink,
} from "styles"

const styles = theme => ({
  container: {
    margin: "auto",
  },
  box: {
    padding: "1px 2px 1px 2px",
    marginTop: "-2px",
  },
})

type Props = {
  /** The annotations data in object form */
  annotations: Object,
  /** Material-UI styling */
  classes: Object,
}

/** Widget that displays the most recent annotations for genes and papers */

const Annotations = (props: Props) => {
  const { annotations, classes } = props

  const genelist = annotations.genes.map((gene, index) => (
    <AnnotationListItems key={index}>{gene}</AnnotationListItems>
  ))

  const paperlist = annotations.papers.map((paper, index) => (
    <AnnotationListItems key={index}>{paper}</AnnotationListItems>
  ))

  return (
    <AnnotationContainer>
      <AnnotationHeader>
        <FontAwesome name="pencil fa-md" />
        <AnnotationTitle>RECENT ANNOTATIONS</AnnotationTitle>
      </AnnotationHeader>
      <Grid container className={classes.container} wrap="wrap">
        <Grid item className={classes.box} xs={6}>
          <AnnotationSubTitle>Genes</AnnotationSubTitle>
          <ListBox margintop="0px" padbottom="0px">
            {genelist}
          </ListBox>
          <MoreLink padbottom="0px">
            <FontAwesome name="plus fa-xs" />
          </MoreLink>
        </Grid>
        <Grid item className={classes.box} xs={6}>
          <AnnotationSubTitle>Papers</AnnotationSubTitle>
          <ListBox margintop="0px" padbottom="0px">
            {paperlist}
          </ListBox>
          <MoreLink padbottom="0px">
            <FontAwesome name="plus fa-xs" />
          </MoreLink>
        </Grid>
      </Grid>
    </AnnotationContainer>
  )
}

export default withStyles(styles)(Annotations)
