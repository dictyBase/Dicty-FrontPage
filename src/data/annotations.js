export default {
    genes: ["abprtA", "trgrB1", "tgrC1", "tacA"],
    papers: ["25609090", "25546705", "25540127", "25596489"]
}