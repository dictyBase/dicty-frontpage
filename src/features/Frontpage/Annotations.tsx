import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

const useStyles = makeStyles({
  mainContainer: {
    textAlign: "center",
    color: "#084b8a",
    backgroundColor: "#e6f2ff",
    padding: "5px 5px 0 5px",
    borderTopLeftRadius: "4px",
    borderTopRightRadius: "4px",
    marginBottom: "10px",
  },
  innerContainer: {
    margin: "auto",
  },
  box: {
    padding: "1px 2px 1px 2px",
    marginTop: "-2px",
  },
  listItem: {
    listStyle: "none",
  },
  title: {
    paddingLeft: "5px",
    fontWeight: "bold",
  },
  header: {
    color: "#0489b1",
    fontSize: "16px",
    borderTopLeftRadius: "5px",
    borderTopRightRadius: "5px",
    backgroundColor: "#fff",
    verticalAlign: "middle",
    textAlign: "center",
    padding: "10px 0 10px 0",
    marginBottom: "10px",

    "@media (maxWidth: 767px)": {
      fontSize: "18px",
      textAlign: "center",
    },
  },
  listBox: {
    padding: "0px 25px 10px 25px",
    fontSize: "12px",
    paddingBottom: "0px",
    marginTop: "0px",
    "@media (max-width: 992px) and (min-width: 767px)": {
      fontSize: "10px",
    },
    "@media (max-width: 768px)": {
      fontSize: "16px",
    },
  },
  updateNotice: {
    color: "rgb(195, 4, 27)",
    fontSize: "11px",
    fontStyle: "italic",
    fontWeight: "normal",
    textAlign: "center",
    paddingBottom: "5px",

    "@media (min-width: 1400px)": {
      paddingTop: "30px",
      fontSize: "12px",
    },
  },
  link: {
    textDecoration: "none",
  },
})

type Props = {
  annotations: {
    genes: string[]
    papers: string[]
  }
}

/** Widget that displays the most recent annotations for genes and papers */

const Annotations = ({ annotations }: Props) => {
  const classes = useStyles()

  const genelist = annotations.genes.map((gene, index) => (
    <li className={classes.listItem} key={index}>
      <a className={classes.link} href={`/gene/${gene}`}>
        {gene}
      </a>
    </li>
  ))

  const paperlist = annotations.papers.map((paper, index) => (
    <li className={classes.listItem} key={index}>
      <a
        className={classes.link}
        href={`https://pubmed.gov/${paper}`}
        target="_blank"
        rel="noopener noreferrer">
        {paper}
      </a>
    </li>
  ))

  return (
    <div className={classes.mainContainer}>
      <div className={classes.header}>
        <FontAwesomeIcon icon="pencil-alt" size="sm" />
        <span className={classes.title}> RECENT ANNOTATIONS</span>
      </div>
      <Grid container className={classes.innerContainer}>
        <Grid item className={classes.box} xs={6}>
          <span className={classes.title}>Genes</span>
          <ul className={classes.listBox}>{genelist}</ul>
        </Grid>
        <Grid item className={classes.box} xs={6}>
          <span className={classes.title}>Papers</span>
          <ul className={classes.listBox}>{paperlist}</ul>
        </Grid>
        <Grid item xs={12}>
          <div className={classes.updateNotice}>updates coming soon</div>
        </Grid>
      </Grid>
    </div>
  )
}

export default Annotations
