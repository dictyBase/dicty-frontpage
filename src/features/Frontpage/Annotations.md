```js
import annotations from "data/annotations"

;<Annotations annotations={annotations} />
```
