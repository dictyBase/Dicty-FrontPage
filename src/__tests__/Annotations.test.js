import React from "react"
import Annotations from "../Components/Annotations"
import renderer from "react-test-renderer"
import annotations from "../././data/annotations"

test("Annotations snapshot test", () => {
  const component = renderer.create(<Annotations annotations={annotations} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
