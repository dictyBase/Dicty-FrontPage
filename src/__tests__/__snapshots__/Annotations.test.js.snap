// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotations snapshot test 1`] = `
<div
  className="sc-eilVRo kEkOqT"
>
  <div
    className="sc-emmjRN MKMGE"
  >
    <span
      aria-hidden={true}
      className="fa fa-pencil fa-md"
    />
    <span
      className="sc-gFaPwZ jLpAvI"
    >
      RECENT ANNOTATIONS
    </span>
  </div>
  <div
    className="sc-fONwsr hprXlC"
  >
    <div
      className="sc-bwzfXH uTxCW sc-bdVaJa hSYntd"
      is={null}
    >
      <span
        className="sc-fhYwyz jYpBTD"
      >
        Genes
      </span>
      <ul
        className="sc-hGoxap dGwfIp"
      >
        <li
          className="sc-jzgbtB kOqXSq"
        >
          abprtA
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          trgrB1
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          tgrC1
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          tacA
        </li>
      </ul>
      <div
        className="sc-krDsej hjRQix"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus fa-xs"
        />
      </div>
    </div>
    <div
      className="sc-bwzfXH uTxCW sc-bdVaJa hSYntd"
      is={null}
    >
      <span
        className="sc-fhYwyz jYpBTD"
      >
        Papers
      </span>
      <ul
        className="sc-hGoxap dGwfIp"
      >
        <li
          className="sc-jzgbtB kOqXSq"
        >
          25609090
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          25546705
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          25540127
        </li>
        <li
          className="sc-jzgbtB kOqXSq"
        >
          25596489
        </li>
      </ul>
      <div
        className="sc-krDsej hjRQix"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus fa-xs"
        />
      </div>
    </div>
  </div>
</div>
`;
